experimental/rsocket-src/transports/TcpConnectionFactory.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/async/EventBaseManager.h>
+#include <folly/io/async/ScopedEventBaseThread.h>
+#include <rsocket/transports/TcpConnectionFactory.h>
+#include "src/framed/FramedDuplexConnection.h"
+#include "src/tcp/TcpDuplexConnection.h"
+
+using namespace ::reactivesocket;
+using namespace ::folly;
+
+namespace rsocket {
+
+// carry the thread/eventbase with the DuplexConnection
+class FramedDuplexConnectionOnThread : public DuplexConnection {
+ public:
+  FramedDuplexConnectionOnThread(
+      std::unique_ptr<FramedDuplexConnection> fd,
+      std::unique_ptr<ScopedEventBaseThread> eventBaseThread)
+      : fd_(std::move(fd)), eventBaseThread_(std::move(eventBaseThread)) {}
+
+  ~FramedDuplexConnectionOnThread() {
+    LOG(INFO) << "FramedDuplexConnectionOnThread => destroy";
+  }
+
+  void setInput(
+      std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> framesSink) {
+    fd_->setInput(std::move(framesSink));
+  }
+
+  std::shared_ptr<Subscriber<std::unique_ptr<folly::IOBuf>>> getOutput() {
+    return fd_->getOutput();
+  }
+
+ private:
+  std::unique_ptr<FramedDuplexConnection> fd_;
+  std::unique_ptr<ScopedEventBaseThread> eventBaseThread_;
+};
+
+// create new ScopedEventBaseThread
+// create new EventBase from it
+// create new AsyncSocket
+// connect
+// create FramedDuplexConnection
+// TODO create variant that takes an existing EventBase
+class SocketConnectorAndCallback : public AsyncSocket::ConnectCallback {
+ public:
+  SocketConnectorAndCallback(OnConnect onConnect, SocketAddress addr)
+      : addr_(addr),
+        onConnect_(onConnect),
+        eventBaseThread_(std::make_unique<ScopedEventBaseThread>()) {}
+
+  ~SocketConnectorAndCallback() {
+    LOG(INFO) << "SocketConnectorAndCallback => destroy";
+  }
+
+  void connect() {
+    // now start the connection asynchronously
+    eventBase_->runInEventBaseThreadAndWait([this]() {
+      LOG(INFO) << "ConnectionFactory => starting socket";
+      socket_.reset(new folly::AsyncSocket(eventBase_));
+
+      LOG(INFO) << "ConnectionFactory => attempting connection to "
+                << addr_.describe() << std::endl;
+
+      socket_->connect(this, addr_);
+
+      LOG(INFO) << "ConnectionFactory  => DONE connect";
+    });
+  }
+
+ private:
+  SocketAddress addr_;
+  folly::AsyncSocket::UniquePtr socket_;
+  OnConnect onConnect_;
+  std::unique_ptr<ScopedEventBaseThread> eventBaseThread_;
+  EventBase* eventBase_{eventBaseThread_->getEventBase()};
```

Comment:
you don't need this member variable, you can just use eventBaseThread_->getEventBase() when you want it.